Running the test program as root inside a sandbox causes access violations

See bug 263585

Patch by Jory A. Pratt

---
--- configure.ac
+++ configure.ac
@@ -315,11 +315,6 @@
 # now tests for mozilla API variance
 # **********************************
 
-# First build a test program to see whether compiling and running XPCOM
-# programs works at all, so we don't get false test failures below
-
-GECKO_XPCOM_PROGRAM_CHECK
-
 # Check for PSM header(s) to infer PSM presence
 
 GECKO_CHECK_HEADERS([pipnss],[nsIX509Cert.h],[have_psm=yes],[have_psm=no])
